lib/internal/child_process.js

```diff
@@ -516,6 +516,12 @@ function setupChannel(target, channel) {
     // a message.
     target._send({ cmd: 'NODE_HANDLE_ACK' }, null, true);
 
+    // It can happen the handle is not received because of some error on
+    // ancillary data reception such as MSG_CTRUNC. In this case don't try to
+    // perform any conversion.
+    if (!handle)
```

Comment:
Yes, as far as I can tell there are two cases where the sending a handle ends up with no handle on reception:
- As Colin said, if the handle is closed while waiting in the sending queue. This case is detected by the sending process and instead of sending a `NODE_HANDLE` message, it sends a *normal* ipc message. In this case, the `sendHandle` argument of the `message` listener callback is set to null.
- When for some reason, the handle cannot be duped on the receiving process. This is the case this PR tries to fix and I think it makes sense that it behaves in the same way as the previous case (setting `sendHandle` to `null`).

I think there's another argument not to treat it as an error. We can send data (the `message` argument) along with the handle that can be useful on reception even if there's no handle associated.